skimage/_shared/tests/test_utils.py

```diff
@@ -347,3 +349,318 @@ def test_deprecate_func():
     )
     assert record[0].lineno == expected_lineno
     assert record[0].filename == __file__
+
+
+@deprecate_parameter("old1", start_version="0.10", stop_version="0.12")
+@deprecate_parameter("old0", start_version="0.10", stop_version="0.12")
+def _func_deprecated_params(arg0, old0=DEPRECATED, old1=DEPRECATED, arg1=None):
+    """Expected docstring.
+
+    Parameters
+    ----------
+    arg0 : int
+        First unchanged parameter.
+    arg1 : int, optional
+        Second unchanged parameter.
+    """
+    return arg0, old0, old1, arg1
+
+
+@deprecate_parameter("old1", new_name="new0", start_version="0.10", stop_version="0.12")
+@deprecate_parameter("old0", new_name="new1", start_version="0.10", stop_version="0.12")
+def _func_replace_params(
+    arg0, old0=DEPRECATED, old1=DEPRECATED, new0=None, new1=None, arg1=None
+):
+    """Expected docstring.
+
+    Parameters
+    ----------
+    arg0 : int
+        First unchanged parameter.
+    new0 : int, optional
+        First new parameter.
+
+        .. versionadded:: 0.10
+    new1 : int, optional
+        Second new parameter.
+
+        .. versionadded:: 0.10
+    arg1 : int, optional
+        Second unchanged parameter.
+    """
+    return arg0, old0, old1, new0, new1, arg1
```

Comment:
I enjoy the brain teaser, but could we have a sequence like:
```suggestion
    return arg0, old0, new0, arg1, old1, new1
```
or
```suggestion
    return arg0, arg1, old0, old1, new0, new1
```
?

Reply:
Same argument, I'd prefer to keep this in the order of the signature / the order they values are passed in. I think that makes the tests somewhat easier to reason about. At least I hope so. :pray: